src/pytest_benchmark/stats.py

```diff
@@ -196,7 +199,8 @@ def __getitem__(self, key):
     def has_error(self):
         return self.fixture.has_error
 
-    def as_dict(self, include_data=True, flat=False, stats=True):
+    def as_dict(self, include_data=True, flat=False, stats=True,
+                cprofile_sort_by="cumtime", cprofile_all_columns=False):
```

Comment:
I would try to reduce the cprofile arguments to a sole `cprofile` argument (cause this call boiler plate is all over the place). If it's specified it means we want to display cprofile stats (sorted by column). If it's not specified it means we want all the columns for storage.

Eg: 

* `.as_dict(cprofile=getoption(...))` if we display the stats in table
* `.as_dict()` if we generate json for storage